src/main/java/edu/umich/ctools/sectionsUtilityTool/Utils.java

```diff
@@ -182,4 +193,56 @@ private static void logMsg(String uniqname, String originalUrl, HttpServletReque
 		M_log.info(String.format(baseString, user, originalUrl));
 	}
 
+	public static String makeApiCall(String... urlChunks){
+		String url = canvasURL + CANVAS_API_VERSION;
+		for (String chunk : urlChunks) {
+			url = url + chunk;
+		}
+		M_log.info("API call: " + url);
+		HttpResponse response = Utils.processApiCall(new HttpGet(url));
+		if (response == null) {
+			M_log.error(String.format("The Api call %s failed with errors", url));
+			return null;
+		}
+		int statusCode = response.getStatusLine().getStatusCode();
+		if (statusCode != HttpStatus.SC_OK) {
+			M_log.error(String.format("The Api call %s has failed with status code %s", url, statusCode));
+			return null;
+		}
+		String apiResponse;
+		String errMsg = "The API call %s has failed to extract the response due to %s";
+		try {
+			apiResponse = EntityUtils.toString(response.getEntity(), "UTF-8");
+		} catch (IOException e) {
+			M_log.error(String.format(errMsg, url, e.getMessage()));
+			return null;
+		} catch (Exception e) {
+			M_log.error(String.format(errMsg, url, e.getMessage()));
+			return null;
+		}
+
+		return apiResponse;
+	}
+
+	public static HttpResponse processApiCall(HttpUriRequest clientRequest) {
```

Comment:
Maybe this should be called executeApiCall?  It would be nice to have names that are more descriptive.  It's not clear from the name what is different between makeApiCall and processApiCall.